enterprise/api/v1/serializers.py

```diff
@@ -214,3 +215,88 @@ def update(self, instance, validated_data):
     user = UserSerializer(read_only=True)
     enterprise_customer = EnterpriseCustomerSerializer(read_only=True)
     data_sharing_consent = UserDataSharingConsentAuditSerializer(many=True, read_only=True)
+
+
+class EnterpriseCatalogSerializer(serializers.Serializer):
+    """
+    Serializer for enterprise customer catalog.
+    """
+    # pylint: disable=invalid-name
+    id = serializers.IntegerField(read_only=True, help_text=_('Course catalog primary key.'))
+    name = serializers.CharField(help_text=_('Catalog name'))
+    query = serializers.CharField(help_text=_('Query to retrieve catalog contents'))
+    courses_count = serializers.IntegerField(read_only=True, help_text=_('Number of courses contained in this catalog'))
+    viewers = serializers.ListField(
+        allow_null=True, allow_empty=True, required=False,
+        help_text=_('Usernames of users with explicit access to view this catalog'),
+        style={'base_template': 'input.html'},
+        child=serializers.CharField(),
+    )
+
+    def create(self, validated_data):
+        """
+        Do not perform any operations for state changing requests.
+        """
+        pass
+
+    def update(self, instance, validated_data):
+        """
+        Do not perform any operations for state changing requests.
+        """
+        pass
+
+
+class EnterpriseCatalogCoursesSerializer(serializers.Serializer):
+    """
+    Serializer for enterprise customer catalog courses.
+    """
+    count = serializers.IntegerField(read_only=True, help_text=_('Total count of catalog courses.'))
+    next = serializers.CharField(read_only=True, help_text=_("URL to fetch next page of courses."))
+    previous = serializers.CharField(read_only=True, help_text=_("URL to fetch previous page of courses."))
+    results = serializers.ListField(read_only=True, help_text=_("list of courses."))
+
+    def update_enterprise_courses(self, request):
+        """
+        This method adds enterprise specific metadata for each course.
+
+        We are adding following fields in all the courses.
+            1. tpa_hint: a string for identifying Identity Provider.
+        """
+        courses = []
+
+        global_context = {
+            'tpa_hint': utils.get_identity_provider_id_for_user(request.user)
+        }
+
+        for course in self.data['results']:
+            courses.append(
+                self.update_course(course, request, global_context)
+            )
+        self.data['results'] = courses
+
+    def update_course(self, course, request, global_context):  # pylint: disable=unused-argument
```

Comment:
I added this function as it would be helpful for future modifications of the serializer.